From 1f7e26a1a843510713c89b4b0d51703de649e0cc Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:36:47 -0400
Subject: [virt] kvm: Add documentation for kvm->srcu lock

Message-id: <1406040016-3289-13-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86091
O-Subject: [RHEL7 PATCH v2 012/221] KVM: Add documentation for kvm->srcu lock
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Thomas Huth <thuth@linux.vnet.ibm.com>

This patch documents the kvm->srcu lock (using the information from
a mail which has been posted by Marcelo Tosatti to the kvm mailing
list some months ago, see the following URL for details:
http://www.mail-archive.com/kvm@vger.kernel.org/msg90040.html )

Signed-off-by: Thomas Huth <thuth@linux.vnet.ibm.com>
Reviewed-by: Xiao Guangrong <xiaoguangrong@linux.vnet.ibm.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 519192aaae38e24d6b32d3d55d791fe294981185)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/virtual/kvm/locking.txt b/Documentation/virtual/kvm/locking.txt
index 41b7ac9..a9f366e 100644
--- a/Documentation/virtual/kvm/locking.txt
+++ b/Documentation/virtual/kvm/locking.txt
@@ -151,3 +151,14 @@ Type:  spinlock_t
 Arch:  any
 Protects: -shadow page/shadow tlb entry
 Comment: it is a spinlock since it is used in mmu notifier.
+
+Name:  kvm->srcu
+Type:  srcu lock
+Arch:  any
+Protects: - kvm->memslots
+  - kvm->buses
+Comment: The srcu read lock must be held while accessing memslots (e.g.
+  when using gfn_to_* functions) and while accessing in-kernel
+  MMIO/PIO address->device structure mapping (kvm->buses).
+  The srcu index can be stored in kvm_vcpu->srcu_idx per vcpu
+  if it is needed by multiple functions.
-- 
1.7.1